src/critnib/critnib.c

```diff
@@ -189,6 +197,14 @@ struct critnib *critnib_new(void) {
  */
 static void delete_node(struct critnib *c, struct critnib_node *__restrict n) {
     if (is_leaf(n)) {
+        if (c->cb_free_leaf && to_leaf(n)) {
```

Comment:
please add a comment that we are calling a callback here

Reply:
Done